docs/tutorials/advanced/performance-optimization.md

```diff
@@ -1,825 +1,513 @@
-
-runner:
-  type: streams
-  config:
-    application.id: optimized-analytics
-    bootstrap.servers: kafka:9092
-
-    # Performance configuration
-    num.stream.threads: 8
-    cache.max.bytes.buffering: 104857600  # 100MB
-    commit.interval.ms: 30000
-
-    # Producer settings
-    producer.linger.ms: 100
-    producer.batch.size: 16384
-
-    # Consumer settings
-    consumer.fetch.max.bytes: 52428800
-    consumer.max.poll.records: 500
+      # Use pre-computed values and avoid object creation
+      priority = value.get("priority", "low")
+      result = efficient_calculation(value.get("amount", 0), priority)
+      return f"processed:{result:.2f}"
 ```
 
-This example:
+Key techniques:
 
-1. Uses Avro for efficient serialization
-2. Implements early filtering to reduce data volume
-3. Processes data in stages for better parallelism
-4. Uses optimized state stores with appropriate caching
-5. Minimizes data copying and transformation
-6. Uses efficient data structures (sets for uniqueness checks)
-7. Limits processing of large collections (top 5 products, top 3 categories)
-8. Includes optimized Kafka Streams configuration
+- **Use globalCode**: Pre-compute expensive operations outside the processing loop
+- **Minimize object creation**: Reuse objects and data structures when possible
+- **Use built-in functions**: They're typically faster than custom implementations
+- **Avoid unnecessary computations**: Compute values only when needed
 
-## Advanced Optimization Techniques
+### Efficient Data Structures
 
-### 1. Custom Serializers/Deserializers
+Choose optimal data structures based on KSML's data type system and your use case:
 
-Implement custom serializers for specialized data types:
+#### KSML Data Types (Performance Ranking)
 
-```yaml
-streams:
-  specialized_data:
-    topic: specialized_events
-    keyType: string
-    valueType: custom
-    serdes:
-      value: org.example.HighPerformanceSerializer
-```
+```text
+# Fastest - Use for high-throughput scenarios
+keyType: string         # Most efficient for keys
+valueType: string       # Fastest serialization
 
-### 2. Memory-Mapped Files
+# Fast - Good balance of performance and functionality  
+keyType: long          # Efficient numeric keys
+valueType: avro:Schema # Efficient binary format with schema
 
-For very large state stores, consider memory-mapped files:
+# Moderate - Flexible but more overhead
+keyType: json          # Flexible structure  
+valueType: json        # Easy to work with, good for Kowl UI
 
-```yaml
-stores:
-  large_state_store:
-    type: custom
-    implementation: org.example.MMapStateStore
-    config:
-      file.path: /data/large-state
-      max.size.bytes: 10737418240  # 10GB
+# Slower - Use only when necessary
+valueType: xml         # XML processing overhead
+valueType: soap        # SOAP envelope overhead
 ```
 
-### 3. Off-Heap Memory
+#### Python Data Structure Guidelines
 
-Use off-heap memory for large state stores:
+```text
```

Comment:
Maybe it's better to brek this up, have the text be normal text and then smaller code blocks marked as python syntax